### PR TITLE
User delete

### PR DESCRIPTION
**Pull request by kushns**
One customer requested functionality to delete user using API so added get_user and delete_user options.

### CODE REVIEW

File: privx_api/privx_api.py [hunk 1]
```diff
@@ -386,6 +402,27 @@ def create_user(self, user: dict) -> PrivXAPIResponse:
         response = self._http_post("userstore.users", body=user)
         return PrivXAPIResponse(response, 201)
 
+    def get_users(self) -> PrivXAPIResponse:
```
Comment (kimmokssh) [inline privx_api/privx_api.py:405]: Might be a bit misleading as this only returns local users. Could change the function name accordingly.

Comment (kushns) [reply privx_api/privx_api.py]: Updated function name to get_local_user

File: privx_api/privx_api.py [hunk 2]
```diff
@@ -250,6 +251,21 @@ def _http_put(self, urlname: str,
 
         return conn.getresponse()
 
+    def _http_delete(self, urlname: str,
+                  body: dict = {},
+                  path_params: dict = {},
+                  query_params: dict = {}) -> http.client.HTTPResponse:
```
Comment (tmorlando) [inline privx_api/privx_api.py:257]: Using mutable types as function default arguments is an anti-pattern every Python programmer has to encounter once. The mutable default arguments are initialized once, and if the values are mutated inside the method, these 'bad' values will be present when the method is next time called. 

The _get_headers() method uses similar antipattern.

If the values are not mutated, no harm is done, but the code is still prone to break easily. 

Demonstration: 
```
>>> def foo(y, x = []):
...  x.append(y)
...  return x
... 
>>> foo(1)
[1]
>>> foo(1)
[1, 1]
```

File: privx_api/privx_api.py [hunk 3]
```diff
@@ -386,6 +402,27 @@ def create_user(self, user: dict) -> PrivXAPIResponse:
...
+        Get users.
+
+        Returns:
+            PrivXAPIResponse
+        """
+        response = self._http_get("userstore.users")
+        return PrivXAPIResponse(response, 200)
+
+    def delete_user(self, user_id: str) -> PrivXAPIResponse:
+        """
```
Comment (tmorlando) [inline privx_api/privx_api.py:416]: for consistency this should be renamed as `delete_local_user()`

Comment (kushns) [reply privx_api/privx_api.py:416]: renamed couple of functions to create_local_user() and delete_local_user()

File: privx_api/privx_api.py (hunk 2 above)
Comment (kushns) [reply privx_api/privx_api.py]: Thanks for reporting the issue, I tried fixing it, please review and share feedback. I am not programmer, just writing small python scripts or modifying existing once.

File: privx_api/privx_api.py [hunk 4]
```diff
@@ -386,6 +415,27 @@ def create_user(self, user: dict) -> PrivXAPIResponse:
         response = self._http_post("userstore.users", body=user)
         return PrivXAPIResponse(response, 201)
 
+    def get_local_users(self) -> PrivXAPIResponse:
```
Comment (kimmokssh) [inline privx_api/privx_api.py:418]: This one is missing the parameters, please check docs.

https://apispecs.ssh.com/#/user/get_local_user_store_api_v1_users

**Comment by kushns** [comment]
> 
> 
> I assume Kimmo meant parameters to control the server side result offset-limit pagination - according to docs the server returns 50 results per request (results 0 - 50) ... and the client is expected request more by providing offset on the subsequent request.

ah.. got it, just updated the code to fix it. so now get_local_user function can be used to get all user and specific user can be used based on username or user_id.

File: privx_api/privx_api.py (hunk 4 above)
Comment (kushns) [reply privx_api/privx_api.py]: @kimmokssh : please check and let me know if all looks fine now.

**Pull request merged**